File: Div2A/CF265-D2-A.cpp
#include <iostream>

int main() {
    int pos(0);
    std::string s;
    std::string t;

    std::cin>>s>>t;

    for (int i = 0; i < t.size(); i++)
    {
        if(t[i] == s[pos])
            pos++;
    }
    
    std::cout<<pos+1;
}

File: Div2A/CF265D2A.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class CF265D2A {
    public static void main(String[] args) throws IOException{
        BufferedReader bi = new BufferedReader(new InputStreamReader(System.in));
        int pos = 0;
        String s = bi.readLine();
        String t = bi.readLine();

        for (int i = 0; i < t.length(); i++) {
            if (t.charAt(i) == s.charAt(pos)) {
                pos++;
            }
        }

        System.out.println(pos+1);
    }
}
